[CI/CD] Claude Code Review Action에서 Bot 트리거 시 실패 문제

## 개요

Claude Code Review GitHub Action이 Bot(claude)이 생성한 이벤트에 의해 트리거될 때 실행이 실패하는 문제입니다.

## 에러 메시지

```
Error: Prepare step failed with error: Workflow initiated by non-human actor: claude (type: Bot). 
Add bot to allowed_bots list or use '*' to allow all bots.
Error: Process completed with exit code 1.
```

## 원인 분석

### 근본 원인
`anthropics/claude-code-action@v1`은 보안상의 이유로 **기본적으로 모든 Bot 트리거를 차단**합니다.

- `allowed_bots` 파라미터 기본값: `""` (빈 문자열 = 모든 봇 차단)
- Claude 봇이 PR을 생성하거나 동기화할 때, 해당 이벤트가 워크플로우를 트리거
- 워크플로우가 Bot에 의해 시작되었으므로 차단됨

### 발생 시나리오
1. Claude Code가 PR 생성 또는 업데이트
2. `pull_request: [opened, synchronize]` 이벤트 발생
3. `claude-code-review.yml` 워크플로우 트리거
4. Action이 트리거 주체가 Bot(claude)임을 감지
5. `allowed_bots`에 claude가 없으므로 실행 거부

## 해결 방안

### 옵션 1: Claude 봇만 허용 (권장)

```yaml
- name: Run Claude Code Review
  uses: anthropics/claude-code-action@v1
  with:
    claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
    allowed_bots: "claude[bot]"  # Claude 봇만 허용
    # ... 기타 설정
```

### 옵션 2: 모든 봇 허용

```yaml
- name: Run Claude Code Review
  uses: anthropics/claude-code-action@v1
  with:
    claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
    allowed_bots: "*"  # 모든 봇 허용
    # ... 기타 설정
```

### 옵션 3: 워크플로우 조건으로 Bot 제외

```yaml
jobs:
  claude-review:
    # Bot이 트리거한 경우 실행하지 않음
    if: github.actor != 'claude[bot]'
    runs-on: ubuntu-latest
```

## 권장 해결책

**옵션 1 (특정 봇만 허용)** 을 권장합니다:
- 보안성 유지: 모든 봇을 허용하는 것보다 안전
- 명시적 허용: Claude 봇의 PR에 대해서도 리뷰 가능

만약 Claude 봇이 생성한 PR에 대해서는 리뷰가 필요 없다면 **옵션 3**도 고려할 수 있습니다.

## 변경 범위

- `.github/workflows/claude-code-review.yml` - `allowed_bots` 파라미터 추가

## 구현 체크리스트

- [ ] `claude-code-review.yml`에 `allowed_bots` 설정 추가
- [ ] 변경 후 Bot이 생성한 PR에서 워크플로우 정상 실행 확인

## 참고 자료

- [claude-code-action 공식 문서](https://github.com/anthropics/claude-code-action)
- [action.yml - allowed_bots 파라미터](https://github.com/anthropics/claude-code-action/blob/main/action.yml)

--- a/.github/workflows/claude-code-review.yml
+++ b/.github/workflows/claude-code-review.yml
@@ -36,6 +36,7 @@ jobs:
         uses: anthropics/claude-code-action@v1
         with:
           claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
+          allowed_bots: "claude[bot]"
           plugin_marketplaces: 'https://github.com/anthropics/claude-code.git'
           plugins: 'code-review@claude-code-plugins'
           prompt: '/code-review:code-review ${{ github.repository }}/pull/${{ github.event.pull_request.number }}'

